packages/contracts/src/dollar/DirectGovernanceFarmer.sol

```diff
@@ -90,42 +98,158 @@ contract DirectGovernanceFarmer is ReentrancyGuard {
+            "Invalid token: must be DAI, USD Coin, Tether, or Ubiquity Dollar"
+        );
         require(amount > 0, "amount must be positive vale");
-        require(durationWeeks >= 1 && durationWeeks <= 208, "duration weeks must be between 1 and 208");
+        require(
+            durationWeeks >= 1 && durationWeeks <= 208,
+            "duration weeks must be between 1 and 208"
+        );
 
         //Note, due to USDT implementation, normal transferFrom does not work and have an error of "function returned an unexpected amount of data"
         //require(IERC20(token).transferFrom(msg.sender, address(this), amount), "sender cannot transfer specified fund");
         IERC20(token).safeTransferFrom(msg.sender, address(this), amount);
-        
+
         address staking = manager.stakingContractAddress();
         address stakingShare = manager.stakingShareAddress();
 
-        uint256 lpAmount;//UAD3CRVf
+        uint256 lpAmount; //UAD3CRVf
         //[Ubiquity Dollar, DAI, USDC, USDT]
         uint256[4] memory tokenAmounts = [
             token == ubiquityDollar ? amount : 0,
             token == token0 ? amount : 0,
             token == token1 ? amount : 0,
             token == token2 ? amount : 0
         ];
-        
+
         //STEP1: add DAI, USDC, USDT or Ubiquity Dollar into metapool liquidity and get UAD3CRVf
         IERC20(token).safeIncreaseAllowance(depositZapUbiquityDollar, amount);
-        lpAmount = IDepositZap(depositZapUbiquityDollar).add_liquidity(ubiquity3PoolLP, tokenAmounts, 0);
+        lpAmount = IDepositZap(depositZapUbiquityDollar).add_liquidity(
+            ubiquity3PoolLP,
+            tokenAmounts,
+            0
+        );
 
         //STEP2: stake UAD3CRVf to Staking
         //TODO approve token to be transferred to Staking contract
         IERC20(ubiquity3PoolLP).safeIncreaseAllowance(staking, lpAmount);
         stakingShareId = IStaking(staking).deposit(lpAmount, durationWeeks);
 
-        IStakingShare(stakingShare).safeTransferFrom(address(this), msg.sender, stakingShareId, 1, '0x');
-
+        IStakingShare(stakingShare).safeTransferFrom(
+            address(this),
+            msg.sender,
+            stakingShareId,
+            1,
+            bytes("")
+        );
         emit Deposit(msg.sender, token, amount, durationWeeks, stakingShareId);
-        
+    }
+
+    /**
+     * @dev Deposit into Ubiquity protocol
+     * @notice Stable coins (DAI / USDC / USDT / Ubiquity Dollar) => uAD3CRV-f => Ubiquity StakingShare
+     * @notice STEP 1 : Change (DAI / USDC / USDT / Ubiquity dollar) to 3CRV at uAD3CRV MetaPool
+     * @notice STEP 2 : uAD3CRV-f => Ubiquity StakingShare
+     * @param tokenAmounts Amount of tokens to deposit (For max: `uint256(-1)`) it MUST follow this order [Ubiquity Dollar, DAI, USDC, USDT]
+     * @param durationWeeks Duration in weeks tokens will be locked (1-208)
+     */
+    function deposit(uint256[4] calldata tokenAmounts, uint256 durationWeeks)
```

Comment:
There's two functions called Deposit. They'll have different signatures but should still have different names.

Reply:
we can have different signatures with the same name. why do you suggest to have another name ? what name do you suggest ?

Reply:
It makes code more readable and can clarify the difference between the two deposit functions without having to read into how they perform differently. 
I would use depositSingle and depositMulti

Reply:
I will update it but can be an interesting discussion for Christmas eve " Pros and Cons of function overloading"

Reply:
> I will update it but can be an interesting discussion for Christmas eve " Pros and Cons of function overloading"

Classic Christmas dinner discussion topics.